static/app/stores/tagStore.tsx

```diff
@@ -171,17 +170,48 @@ const storeConfig: TagStoreDefinition = {
   },
 
   loadTagsSuccess(data) {
-    const newTags = data.reduce<TagCollection>((acc, tag) => {
-      acc[tag.key] = {
+    // Note: We could probably stop cloning the data here and just
+    // assign to this.state directly, but there is a change someone may
+    // be relying on referential equality somewhere in the codebase and
+    // we dont want to risk breaking that.
+    const newState = {};
+
+    for (let i = 0; i < data.length; i++) {
```

Comment:
We dont check max store size here which I think is ok as 1k is the max that we receive API anyways and it probably makes sense to store all the tags we receive from the API so that they are not just magically dropped somewhere.